lib/permissions.js

```diff
@@ -7,16 +7,36 @@
 
 var vocab = require('./vocab')
 var rdf = require('./util/rdf-parser').rdflib
-var auth = require('./auth')
 var webClient = require('./web')
+var path = require('path')
+var md5 = require('md5')
+
+var inheritancePredicateName = 'defaultForNew'
 
 var PermissionType = {
-  Owner: 'Owner',
+  Control: 'Control',
   Read: 'Read',
   Write: 'Write',
   Append: 'Append'
 }
 
+// Represents an Agent object to distinguish singular agents vs user groups
+function Agent (agentURI, isSingular) {
+  this.uri = agentURI || ''
+  this.isSingular = typeof isSingular !== 'undefined' ? isSingular : true
```

Comment:
This is typically written as:

```js
this.isSingular = isSingular
```
If the `isSingular` parameter is not passed into the constructor, the *value* of that variable will be `undefined`, which will evaluate as a false-y value in any sort of if/then statements. Meaning, just assigning it to the object property is fine, no need to safeguard with `typeof`.

The reason people use `typeof x === 'undefined'` is in case `x` has not been *defined* as a variable. But a function parameter can never be undefined as a variable -- it's defined right there in the function definition. A function parameter can have a *value* of `undefined`, but there's no need to safeguard against that, you can just go ahead and use it like:  

```js
if (agent.isSingular) {  // do whatever }
```

Does that make sense?